src/main/java/com/cloudbees/springboot_fm_example/controller/DemoController.java

```diff
@@ -0,0 +1,81 @@
+package com.cloudbees.springboot_fm_example.controller;
+
+import com.cloudbees.springboot_fm_example.container.Flags;
+import com.cloudbees.springboot_fm_example.service.DemoService;
+import io.rollout.client.NetworkConfigurationsOptions;
+import io.rollout.rox.server.Rox;
+import io.rollout.rox.server.RoxOptions;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.concurrent.ExecutionException;
+
+@RestController
+@RequestMapping("/api")
+public class DemoController implements InitializingBean {
+
+    Flags flags;
+    DemoService demoService;
+
+    @Value("${rox.key:}")
+    private String environmentKey;
+    @Value("${rox.options.getConfigApiEndpoint:}")
+    private String getConfigApiEndpoint;
+    @Value("${rox.options.getConfigCloudEndpoint:}")
+    private String getConfigCloudEndpoint;
+    @Value("${rox.options.sendStateApiEndpoint:}")
+    private String sendStateApiEndpoint;
+    @Value("${rox.options.sendStateCloudEndpoint:}")
+    private String sendStateCloudEndpoint;
+    @Value("${rox.options.analyticsEndpoint:}")
+    private String analyticsEndpoint;
+    @Value("${rox.options.pushNotificationsEndpoint:}")
+    private String pushNotificationsEndpoint;
+
+    @Autowired
+    public DemoController(
+            Flags flags,
+            DemoService demoService) {
+        this.flags = flags;
+        this.demoService = demoService;
+    }
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        initFlags();
+    }
+
+    public void initFlags() {
+        if (environmentKey.equals("<INSERT YOUR SDK KEY HERE>"))
+            throw new RuntimeException("You haven't yet inserted your SDK Key into FeatureFlagsProvider.tsx - the application below will not update until you do so. Please check the README.md for instructions.");
+
+        RoxOptions options = new RoxOptions.Builder()
+                .withDisableSignatureVerification(true)
+                .withNetworkConfigurationsOptions(
```

Comment:
Why do we need to configure all these? Is it because Java does not yet support cloudbees.io? (I thought it was already done, so either it works & you shouldn't need _any_ additional configuration, or it doesn't and we need to fix it :-) ).

Reply:
Java was not working without specifying those. I checked with Asaf and said it looked ok.

Reply:
@AsafRollout - do you know why we need these parameters? I thought Java SDK had already had platform support (auto-switching on uuid/mongoid) as it was one of our primary SDKs.
(& its not even in the [SDK parity epic](https://cloudbees.atlassian.net/browse/CBP-626), because supposedly it already supports CBP:
> Like previous Python, JavaScript, Go, and Java SDK work that has already been done for the platform.

Reply:
@AsafRollout? I'm worried we're missing some functionality on the Java SDK; I thought this was all done (since one of the primary four).